doc/cookbook/jwt-and-basic-auth/JWTAndBasicAuth.lhs

```diff
@@ -0,0 +1,168 @@
+# Combining JWT-based authentication with basic access authentication
+
+In this example we will make a service with basic HTTP authentication
+for Haskell clients and other programs, as well as with JWT-based
+authentication for web browsers. Web browsers will still use basic
+HTTP authentication to retrieve JWT tokens though.
+
+While basic authentication comes with Servant itself,
+[servant-auth](https://hackage.haskell.org/package/servant-auth) and
+[servant-auth-server](https://hackage.haskell.org/package/servant-auth-server)
+packages are needed for the JWT-based one.
+
+This recipe uses the following ingredients:
+
+```haskell
+{-# LANGUAGE OverloadedStrings, TypeFamilies, DataKinds,
+  DeriveGeneric, TypeOperators #-}
+import Data.Aeson
+import GHC.Generics
+import Data.Proxy
+import System.IO
+import Network.HTTP.Client (newManager, defaultManagerSettings)
+import Network.Wai.Handler.Warp
+import Servant as S
+import Servant.Client
+import Servant.Auth as SA
+import Servant.Auth.Server as SAS
+import Control.Monad.IO.Class (liftIO)
+
+port :: Int
+port = 3001
+```
+
+
+## Authentication
+
+That's how we'll represent a user: usually user identifier is handy to
```

Comment:
"Below is how  [...]" might be more natural here for the reader, as the example only comes _after_ this sentence.